### ISSUE TITLE
False negative with bound type variable

### ISSUE THREAD

**Issue by hauntsaninja** [astral-sh/ty#2408]
### Summary

```python
from typing import *

class Base: ...

BaseT = TypeVar("BaseT", bound=Base)

def f(x: Callable[[], BaseT]) -> BaseT:
    return x()

class Unrelated: ...

def make_unrelated() -> Unrelated:
    return Unrelated()

f(Unrelated)  # should error
f(make_unrelated)  # should error
```

I have a PR https://github.com/astral-sh/ruff/pull/19946 that solves this, but is not mergeable. Filing this as a way of tracking this false negative, in case my PR never makes it

**Comment by carljm** [comment]
Thanks! It seems like https://github.com/astral-sh/ruff/pull/19946 should not be needed to solve this, so there's another bug to be identified here.

**Comment by charliermarsh** [comment]
It looks like this is also fixed by https://github.com/astral-sh/ruff/pull/26964.
